src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -1223,75 +1221,90 @@ bool entitiesQuery
           return true;
         }
 
-        for (unsigned int ix = 0; ix < queryAttrV.size(); ++ix) {
+        std::set<std::string> attrNames;
+        queryAttrs.getFieldNames(attrNames);
+        for( std::set<std::string>::iterator i = attrNames.begin(); i!=attrNames.end(); ++i)
```

Comment:
I have tried to fix it in the overall code (some changes in part outside this PR).